### ISSUE TITLE
Trouble to find tests in sub folders 

### ISSUE THREAD

**Issue by ybenouag** [mathworks/jenkins-matlab-plugin#17]
Hello,

I'm actually using Matlab, Git Lab and Jenkins in order to have a CI environment for my Matlab projects. I usually have an architecture like this one where my test folder is a sub folder of a sub folder: 
![architecture](https://user-images.githubusercontent.com/48535945/56004353-239e1a00-5d0e-11e9-81e4-fe0d13706618.PNG)

Before I found your plugin I was using a batch script who was running my runAllTests script using the same workflow as the script generated by your plugin. But the things is that the testsuite function generated can't find my tests .

![Generated code](https://user-images.githubusercontent.com/48535945/56004350-23058380-5d0e-11e9-9185-cbb59d54d021.PNG)

![result in powershell](https://user-images.githubusercontent.com/48535945/56004354-239e1a00-5d0e-11e9-9ed4-db50a99ef2cb.PNG)

I'm not a Matlab pro but the only solution that a find at the moment is to change the testsuite function by using the TestSuite object of the Matlab framework. 

![Result in Powershell after changed](https://user-images.githubusercontent.com/48535945/56004352-23058380-5d0e-11e9-8aaa-e9148903442a.PNG)

![Code changes](https://user-images.githubusercontent.com/48535945/56004351-23058380-5d0e-11e9-83cb-0786713f7adb.PNG)

This was by using the automatic configuration of the plugin. Actually, I'm forced to use the custom but, if this change won't brake anything , maybe it can be a better way to generate the script that run all the tests in a project. 

Otherwise great job guys 💯 

Yannis Ben Ouaghrem.

**Comment by acampbel** [comment]
Hi Yannis,

Thank you for creating this issue! This is actually by design, but I think it would be great to get your feedback and how you experienced the behavior. Basically, what is happening here is that since you are using the new MATLAB project feature, running the testsuite function in the project root folder will use the new TestSuite.fromProject method for creating the suite. This function creates the suite from all the files classified with the "Test" classification in the project metadata as the suite. In this case none of your tests were classified as Test so none of them ran. Ultimately, we envision this will be a much better and robust specification of what is test content in a project, and there are many other project features such as environment setup, path specification and so forth, which will make the automatic option compelling. 

However, for now it seems that this was not obvious. Also, we would love to hear from you regarding how this works for you and what pains may exist in this workflow.

Thanks!

**Comment by ybenouag** [comment]
Hi Andy, 
Thanks for the reply you were right the problem come from a bad metadata initialisation on my project ressources. Only my classes based unit test were categorized as test and my function and script based unit test were categorized as design. By changing manually those categories it's now working perfectly.

Thanks a lot!

**Comment by acampbel** [comment]
Sounds great. How does this work for you? Is this a rational/easy method for ensuring the right tests run? Also interested to hear if there was something we could have done differently to perhaps give the hint of what was going on.

**Comment by ybenouag** [comment]
I'm an intern at the Mathworks office in Sydney ,and I made and presented  a demo, showing how to run Unit Test in Matlab. I also showed how to configure Jenkins by following articles of your blog. So before your plugin, I had my own runAllTest script that ran all the scripts of the project. All the results generated by the plugins were stored in a specific folder. 
Technically, your plugin simplifies a lot this procedure and make a lot easier the configuration of Jenkins . So yes, this is the best way to run matlab unit test on a Jenkins server. For me as well, all was clear just by going through the readme.md. I just didn't know that the tests were categorized as a test category in the Project functionalities.
